vsintegration/src/FSharp.Editor/Navigation/OpenDeclarationsGetter.fs

```diff
@@ -0,0 +1,481 @@
+                             Parent = from |> Option.map (fun from -> from.Split '.') }
+                }))
+
+    let updateOpenDeclsWithSymbolPrefix symbolPrefix symbolRange openDecls = 
+        openDecls 
+        |> List.fold (fun (acc, foundMatchedDecl) openDecl -> 
+            // We already found a matched open declaration or the symbol range is out or the next open declaration.
+            // Do nothing, just add the open decl to the accumulator as is.
+            if foundMatchedDecl || not (Range.rangeContainsRange openDecl.ScopeRange symbolRange) then
+                openDecl :: acc, foundMatchedDecl
+            else
+                let matched, updatedDecl = openDecl |> OpenDeclaration.updateBySymbolPrefix symbolPrefix
+                updatedDecl :: acc, matched
+            ) ([], false)
+        |> fst
+        |> List.rev
+
+    let setOpenDeclsIsUsedFlag idents (openDecls: OpenDeclaration list) =
+        openDecls
+        |> List.map (fun decl -> 
+            let declarations = 
+                decl.Declarations 
+                |> List.map (fun d -> 
+                    if d.IsUsed then d
+                    else { d with IsUsed = d.Declarations |> List.exists (Array.areEqual idents) })
+            let isUsed = declarations |> List.exists (fun d -> d.IsUsed)
+            { decl with Declarations = declarations; IsUsed = isUsed })
+
+    let spreadIsUsedFlagToParents (openDecls: OpenDeclaration list) =
+        let res = 
+            openDecls 
+            |> List.fold (fun res decl ->
+                if not decl.IsUsed then res
+                else
+                    let parents = decl.Declarations |> List.choose (fun decl -> decl.Parent)
+                    parents 
+                    |> List.fold (fun res parent -> res |> setOpenDeclsIsUsedFlag parent)
+                       res
+               ) openDecls
+        //debug "[OpenDeclarationsGetter] spreadIsUsedFlagToParents: Before = %A, After = %A" openDecls res
+        res
+
+    type Line = int
+    type EndColumn = int
+
+    let private processOpenDeclaration (longIdent: Ident list) =
+        let isExactlyOne =
+            match longIdent with
+            | [_] -> true
+            | _ -> false
+        longIdent
+        |> List.mapi (fun i ident -> 
+            // Only filter out if global is a prefix of the open declaration
+            match i, ident.idText with
+            | 0, "`global`" when not isExactlyOne ->
+                let r = ident.idRange
+                // Make sure that we don't filter out ``global`` and the like
+                if r.StartLine = r.EndLine && r.EndColumn - r.StartColumn = 6 then
+                    None
+                else Some ident
+            | _ -> 
+                Some ident)
+        |> List.choose id
+        |> longIdentToArray
+
+    let getOpenDeclarations (ast: ParsedInput option) (entities: AssemblySymbol list option) 
```

Comment:
This functions is not used anywhere. Why have you copied it? Please, examine all the functions in this file (and in the entire PR) for usage.